extensions/amp-viewer-integration/0.1/messaging.js

```diff
@@ -48,6 +49,20 @@ export let Message;
 export let RequestHandler;
 
 /**
+  * @param {*} message
+  * @return {?Message}
+  */
+export function parseMessage(message, isWebview_) {
+  if (!isWebview_) {
+    return message;
+  }
+  if (!(typeof message == 'string') || message.charAt(0) != '{') {
+    return null;
+  }
+  return tryParseJson(message);
```

Comment:
This returns `undefined` in case of failure. Since you define result as nullable, we should probably do `return tryParseJson(message) || null;`.